alphazero/model.py

```diff
@@ -0,0 +1,181 @@
+        out = self.conv1(x)
+        if self.bn1 is not None:
+            out = self.bn1(out)
+        out = self.activation(out)
+
+        # second convolution, add input after batchnorm step
+        out = self.conv2(out)
+        if self.bn2 is not None:
+            out = self.bn2(out)
+        out += residual
+        out = self.activation(out)
+
+        return out
+
+
+class SidNet(nn.Module):
+    """
+    Creates the Encoding Neural Network. Takes
+            block: type of blocks to construct.(ResidualBlocks)
+            in_channels: # of game states(previous turns and positions)
+            enc_channels: number of channels to encode
+            num_blocks: number of residual blocks to create
+    """
+
+    def __init__(
+        self,
+        block,
+        in_channels: int,
+        enc_channels: int,
+        num_blocks: int,
+        activation=nn.ReLU(inplace=False),
+        batch_on=True,
+    ):
+        super(SidNet, self).__init__()
+        self.conv1 = nn.Conv2d(
+            in_channels, enc_channels, kernel_size=3, stride=1, padding=1
+        )
+        self.bn = nn.BatchNorm2d(enc_channels) if batch_on else None
+        self.resblock = self.make_layer(block, enc_channels, num_blocks)
+        self.activation = activation
+
+    def make_layer(self, block, out_channels, num_blocks):
+
+        layers = []
+        for i in range(0, num_blocks):
+            layers.append(block(out_channels, out_channels))
+        return nn.Sequential(*layers)
+
+    def forward(self, x):
+
+        encoding = self.conv1(x)
+        if self.bn is not None:
+            encoding = self.bn(encoding)
+        encoding = self.activation(encoding)
+        encoding = self.resblock(encoding)
+
+        return encoding
+
+
+class PolicyHead(nn.Module):
+    """
+    Create the PolicyHead Network, which take the encoded features and the game
+    being played from the previous SidNet Network and returns a vector of
+    length game.width * game.height of move probabilities for each specific
+    position. Following the architecture of the AlphaGoZero Paper, requires the
+    number of channels from the previous residual nets, the game being played,
+    the activation function, and if bathcnorm is on or off.
+    """
+
+    def __init__(
+        self,
+        enc_channels: int,
+        game,
+        activation=nn.ReLU(inplace=False),
+        batch_on=True,
+        pass_move=False,
+    ):
+
+        super(PolicyHead, self).__init__()
+        self.policyconv = nn.Conv2d(
+            enc_channels, out_channels=2, kernel_size=1, stride=1
+        )
+        self.policybn = nn.BatchNorm2d(2) if batch_on else None
+
+        self.game_width = game.width
+        self.game_height = game.height
+        self.board_pos = game.width * game.height
+        self.pass_move = pass_move
+        if self.pass_move:
+            self.policyfc = nn.Linear(
+                in_features=self.board_pos * 2, out_features=self.board_pos + 1,
```

Comment:
Since `self.pass_move` is supposed to be `True` or `False` you can just add it to `self.board_pos` and it should convert to 1/0 appropriately. no need for this if/else